ee/src/main/java/org/jboss/as/ee/subsystem/ManagedScheduledExecutorServiceAdd.java

```diff
@@ -76,24 +72,26 @@ protected void performRuntime(OperationContext context, ModelNode operation, Mod
         final AbstractManagedExecutorService.RejectPolicy rejectPolicy = AbstractManagedExecutorService.RejectPolicy.valueOf(ManagedScheduledExecutorServiceResourceDefinition.REJECT_POLICY_AD.resolveModelAttribute(context, model).asString());
 
         final ManagedScheduledExecutorServiceService service = new ManagedScheduledExecutorServiceService(name, jndiName, hungTaskThreshold, longRunningTasks, coreThreads, keepAliveTime, keepAliveTimeUnit, threadLifeTime, rejectPolicy);
-        final ServiceBuilder<ManagedScheduledExecutorServiceAdapter> serviceBuilder = context.getServiceTarget().addService(ConcurrentServiceNames.getManagedScheduledExecutorServiceServiceName(name), service);
+        final ServiceBuilder serviceBuilder = context.getCapabilityServiceTarget()
+                .addService(ManagedScheduledExecutorServiceResourceDefinition.CAPABILITY.getCapabilityServiceName(name))
+                .setInstance(service);
 
         String contextService = null;
         if(model.hasDefined(ManagedScheduledExecutorServiceResourceDefinition.CONTEXT_SERVICE)) {
             contextService = ManagedScheduledExecutorServiceResourceDefinition.CONTEXT_SERVICE_AD.resolveModelAttribute(context, model).asString();
         }
         if (contextService != null) {
-            serviceBuilder.addDependency(ConcurrentServiceNames.getContextServiceServiceName(contextService), ContextServiceImpl.class, service.getContextServiceInjector());
+            serviceBuilder.addDependency(context.getCapabilityServiceName(ContextServiceResourceDefinition.CAPABILITY.getName(), contextService, ContextServiceImpl.class), ContextServiceImpl.class, service.getContextServiceInjector());
         }
         String threadFactory = null;
         if(model.hasDefined(ManagedScheduledExecutorServiceResourceDefinition.THREAD_FACTORY)) {
             threadFactory = ManagedScheduledExecutorServiceResourceDefinition.THREAD_FACTORY_AD.resolveModelAttribute(context, model).asString();
         }
         if (threadFactory != null) {
-            serviceBuilder.addDependency(ConcurrentServiceNames.getManagedThreadFactoryServiceName(threadFactory), ManagedThreadFactoryImpl.class, service.getManagedThreadFactoryInjector());
+            serviceBuilder.addDependency(context.getCapabilityServiceName(ManagedThreadFactoryResourceDefinition.CAPABILITY.getName(), threadFactory, ManagedThreadFactoryImpl.class), ManagedThreadFactoryImpl.class, service.getManagedThreadFactoryInjector());
         }
-        if(rcPresent) {
-            serviceBuilder.addDependency(RequestController.SERVICE_NAME, RequestController.class, service.getRequestController());
+        if (context.hasOptionalCapability(REQUEST_CONTROLLER_CAPABILITY_NAME, null, null)) {
+            serviceBuilder.addDependency(context.getCapabilityServiceName(REQUEST_CONTROLLER_CAPABILITY_NAME, RequestController.class), RequestController.class, service.getRequestController());
```

Comment:
Here, you would want to use (note that serviceBuilder is a CapabilityServiceBuilder<?>):
```java
  serviceBuilder.addCapabilityRequirement(REQUEST_CONTROLLER_CAPABILITY_NAME, RequestController.class, service.getRequestController());
```

Reply:
done